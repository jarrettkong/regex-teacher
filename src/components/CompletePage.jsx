import React from 'react'

const CompletePage = (props) => {
  return (
    <div className="Lesson-complete-container">
      <h1 className="Lesson-complete-text">Congratulations</h1>
      <p>You have completed the course, if you would like more practice, you can click the reset button to start over.</p>
      <button className="App-btn Lesson-btn" id="resetButton" onClick={props.resetProgress}>Restart</button>
    </div>
  )
}

export default CompletePage;