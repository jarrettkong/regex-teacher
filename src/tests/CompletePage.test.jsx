import React from 'react';
import CompletePage from '../components/CompletePage.jsx';
import { shallow } from 'enzyme';

const mockResetProgress = jest.fn();

describe('Home', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
      <CompletePage resetProgress={mockResetProgress} />
    )
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('resetButton', () => {
    it('should call resetProgress on click', () => {
      wrapper.find('#resetButton').simulate('click', { preventDefault: () => {} })
      expect(mockResetProgress).toHaveBeenCalled();
    })
  })
  
})