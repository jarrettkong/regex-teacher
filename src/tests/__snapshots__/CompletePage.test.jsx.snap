// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<div
  className="Lesson-complete-container"
>
  <h1
    className="Lesson-complete-text"
  >
    Congratulations
  </h1>
  <p>
    You have completed the course, if you would like more practice, you can click the reset button to start over.
  </p>
  <button
    className="App-btn Lesson-btn"
    id="resetButton"
    onClick={[MockFunction]}
  >
    Restart
  </button>
</div>
`;
